cpp/ql/src/semmle/code/cpp/models/interfaces/Allocation.qll

```diff
@@ -85,3 +85,16 @@ abstract class AllocationExpr extends Expr {
    */
   predicate requiresDealloc() { any() }
 }
+
+/**
+ * An `operator new` or `operator new[]` function that may be associated with
+ * `new` or `new[]` expressions.  Note that `new` and `new[]` are not function
+ * calls, but these functions may also be called directly.
+ */
+abstract class OperatorNewAllocationFunction extends AllocationFunction {
```

Comment:
Do we expect there to ever be a new override of this class? It seems to me that the description of the abstract class covers exactly the functions that the implementation covers. It might be a better candidate for `commons`, or we could figure out how to arrange public vs private classes within the implementation directory.

Reply:
No we don't expect there to ever be another override - but it needs to be abstract if its to have no characteristic predicate in the interface.  I could just put the full implementation here or, as you say, in `commons` (but that would spread out things to do with allocation even more).

Another option would be to replace this interface with a predicate or predicates in the parent class (e.g. `isOperatorNewAllocationFunction()`).

Reply:
I'd prefer a public concrete implementation over an abstract class that we don't want overridden. Putting it here probably strikes the best balance.

Reply:
Done.

Reply:
I suppose that it's possible that some codebases want to model their custom overloads of `operator new()` specially. I don't feel strongly about that scenario, though, so I don't think there's a need to change this without a concrete scenario.